ctsm_filenames.csv

```diff
@@ -40,29 +40,29 @@
 50,/glade/scratch/negins/ctsm_0110/src/cpl/mct,lnd_import_export.F90,lnd_import_export.F90,lnd_import_export,lnd_import_export
 51,/glade/scratch/negins/ctsm_0110/src/cpl/mct,lnd_comp_mct.F90,lnd_comp_mct.F90,lnd_comp_mct,lnd_comp_mct
 52,/glade/scratch/negins/ctsm_0110/src/cpl/nuopc,lnd_import_export.F90,lnd_import_export.F90,lnd_import_export,lnd_import_export
 53,/glade/scratch/negins/ctsm_0110/src/cpl/nuopc,lnd_shr_methods.F90,lnd_shr_methods.F90,lnd_shr_methods,lnd_shr_methods
 54,/glade/scratch/negins/ctsm_0110/src/cpl/nuopc,lnd_comp_nuopc.F90,lnd_comp_nuopc.F90,lnd_comp_nuopc,lnd_comp_nuopc
-55,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/main,GetGlobalValuesMod_stub.F90,ctsm_GetGlobalValuesStub.F90,GetGlobalValuesMod_stub,ctsm_GetGlobalValuesStub
-56,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/main,histFileMod_stub.F90,ctsm_HistFileStub.F90,histFileMod_stub,ctsm_HistFileStub
+55,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/main,GetGlobalValuesMod_stub.F90,ctsm_GetGlobalValues_stub.F90,GetGlobalValuesMod,ctsm_GetGlobalValues
+56,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/main,histFileMod_stub.F90,ctsm_HistFile_stub.F90,histFileMod,ctsm_HistFile
 57,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/main,ncdio_dim.F90,ctsm_NcdIoDim.F90,ncdio_dim,ctsm_NcdIoDim
-58,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/csm_share,seq_comm_mct.F90,ctsm_SeqCommMct.F90,seq_comm_mct,ctsm_SeqCommMct
-59,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/csm_share,shr_mpi_mod_stub.F90,ctsm_ShrMpiStub.F90,shr_mpi_mod_stub,ctsm_ShrMpiStub
-60,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/csm_share,mct_mod_stub.F90,ctsm_MctStub.F90,mct_mod_stub,ctsm_MctStub
-61,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/utils,clmfates_paraminterfaceMod_stub.F90,ctsm_FatesParamInterfaceStub.F90,clmfates_paraminterfaceMod_stub,ctsm_FatesParamInterfaceStub
-62,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/utils,spmdMod_stub.F90,ctsm_SpmdStub.F90,spmdMod_stub,ctsm_SpmdStub
-63,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/dyn_subgrid,dynFileMod_stub.F90,ctsm_DynFileStub.F90,dynFileMod_stub,ctsm_SpmdStub
+58,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/csm_share,seq_comm_mct.F90,seq_comm_mct.F90,seq_comm_mct,seq_comm_mct
+59,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/csm_share,shr_mpi_mod_stub.F90,shr_mpi_mod_stub.F90,shr_mpi_mod,shr_mpi_mod
+60,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/csm_share,mct_mod_stub.F90,mct_mod_stub.F90,mct_mod,mct_mod
+61,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/utils,clmfates_paraminterfaceMod_stub.F90,ctsm_FatesParamInterface_stub.F90,CLMFatesParamInterfaceMod,ctsm_FatesParamInterface
+62,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/utils,spmdMod_stub.F90,ctsm_Spmd_stub.F90,spmdMod,ctsm_Spmd
+63,/glade/scratch/negins/ctsm_0110/src/unit_test_stubs/dyn_subgrid,dynFileMod_stub.F90,ctsm_DynFile_stub.F90,dynFileMod,ctsm_DynFile

```

Comment:
This stub stuff deserves some explanation of my thinking. A key thing to understand here is that these stub modules are compiled in place of the real ones via build-time logic in the unit test build. So it is critical that the module name be the same as the real module name: we can't have "stub" on the module name. As for the file names: I don't feel strongly about the changes I suggested here, but I have a general rule that, when there is a part of a name that has a different characteristic from the rest of the name, then the separator/convention for that part should differ. That's pretty vague. What I mean here is, for example: We have a file named `ctsm_GetGlobalValues.F90` and then another file that is a stub of that one. In the stub's file name, I wanted the base name (`ctsm_GetGlobalValues`) to be grouped together, and the fact that it's a stub to be separated from that base name. I felt that `ctsm_GetGlobalValues_stub` accomplished this goal better than `ctsm_GetGlobalValuesStub`. But again, I don't feel strongly about this.

Reply:
Also, some of these are stubs of share code, which doesn't get the `ctsm_` prefix.